Use of unicode_internals points people to issue #0

There is no issue #0 :-) I'm guessing someone intended #0 to mean "we don't have an issue for this", but either the error message emitting machinery needs to suppress it, or it should be given a real error message. Here's what it looks like:

```
error[E0658]: use of unstable library feature 'unicode_internals' (see issue #0)
  --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-syntax-94.0.0/parse/lexer/mod.rs:18:5
   |
18 | use std_unicode::property::Pattern_White_Space;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(unicode_internals)] to the crate attributes to enable
error[E0658]: use of unstable library feature 'unicode_internals' (see issue #0)
    --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-syntax-94.0.0/parse/lexer/mod.rs:1725:21
     |
1725 |     c.map_or(false, Pattern_White_Space)
     |                     ^^^^^^^^^^^^^^^^^^^
     |
     = help: add #![feature(unicode_internals)] to the crate attributes to enable
error: aborting due to 2 previous errors
```

You can see it here: https://travis-ci.org/alex/ct-tools/builds/366818424

--- a/src/libsyntax/feature_gate.rs
+++ b/src/libsyntax/feature_gate.rs
@@ -1228,10 +1228,9 @@ fn leveled_feature_err<'a>(sess: &'a ParseSess, feature: &str, span: Span, issue
         GateIssue::Library(lib) => lib,
     };
 
-    let explanation = if let Some(n) = issue {
-        format!("{} (see issue #{})", explain, n)
-    } else {
-        explain.to_owned()
+    let explanation = match issue {
+        None | Some(0) => explain.to_owned(),
+        Some(n) => format!("{} (see issue #{})", explain, n)
     };
 
     let mut err = match level {

--- a/src/test/ui/feature-gate/issue-49983-see-issue-0.rs
+++ b/src/test/ui/feature-gate/issue-49983-see-issue-0.rs
@@ -1,0 +1,16 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+extern crate core;
+
+// error should not say "(see issue #0)"
+#[allow(unused_imports)] use core::ptr::Unique; //~ ERROR use of unstable library feature
+
+fn main() {}

--- a/src/test/ui/feature-gate/issue-49983-see-issue-0.stderr
+++ b/src/test/ui/feature-gate/issue-49983-see-issue-0.stderr
@@ -1,0 +1,11 @@
+error[E0658]: use of unstable library feature 'ptr_internals': use NonNull instead and consider PhantomData<T> (if you also use #[may_dangle]), Send, and/or Sync
+  --> $DIR/issue-49983-see-issue-0.rs:14:30
+   |
+LL | #[allow(unused_imports)] use core::ptr::Unique; //~ ERROR use of unstable library feature
+   |                              ^^^^^^^^^^^^^^^^^
+   |
+   = help: add #![feature(ptr_internals)] to the crate attributes to enable
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0658`.

